src/components/tabs/Tabs.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <div class="tabs">
+    <ul class="tabs__controls" role="tablist">
+      <li
+        class="tabs__control"
+        role="presentation"
+        v-for="tab in tabs"
+        v-bind:key="tab.id"
+      >
+        <button
+          type="button"
+          role="tab"
+          v-on:click="changeTab(tab.id)"
+          :class="['tabs__button', { 'is-active': activeTabId === tab.id }]"
+          :id="tab.id + '-tab'"
+          :aria-controls="tab.id"
+          :aria-selected="activeTabId === tab.id"
+        >
+          {{ tab.label }}
+        </button>
+      </li>
+    </ul>
+
+    <div class="tabs__item">
+      <div
+        role="tabpanel"
+        tabindex="0"
+        v-for="tab in tabs"
+        v-bind:key="tab.id"
+        :class="['tabs__content', { 'is-active': activeTabId === tab.id }]"
+        :id="tab.id"
+        :aria-labelledby="tab.id + '-tab'"
+      >
+        {{ tab.content }}
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import type { TabItemProps } from 'src/models/Tabs'
+
+export default defineComponent({
+  name: 'Tabs',
+  data() {
+    return {
+      activeTabId: this.defaultTabId || this.tabs[0].id,
```

Comment:
do you think it's worth adding a check here to avoid `index is out of range` or `object is undefined` errors? In case `tabs` is undefined or empty

Reply:
Yes for sure! I've now added a custom validator for tabs for a bit more safety, and also added support for optional chaining to avoid those errors on the fallback value.